### PR TITLE
Rekovic hlt cmssw 8 0 x 2016 02 11 2300

### PR DESCRIPTION
**Pull request by rekovic**
For  TESTING.
Note: Refs to Stage2 L1T  stored in the interface  and TriggerFilterObjectIWithRefs retains them in the Event.

### CODE REVIEW

File: DataFormats/HLTReco/interface/TriggerRefsCollections.h [hunk 1]
```diff
@@ -385,13 +393,11 @@ namespace trigger
       return l1tjetIds_.size();
     }
     size_type addObjects (const Vids& ids, const VRl1ttau& refs) {
-      assert(ids.size()==refs.size());
       l1ttauIds_.insert(l1ttauIds_.end(),ids.begin(),ids.end());
```
Comment (Martin-Grunewald) [inline DataFormats/HLTReco/interface/TriggerRefsCollections.h]: Why do you remove the assert?

File: DataFormats/HLTReco/interface/TriggerRefsCollections.h [hunk 2]
```diff
@@ -385,13 +393,11 @@ namespace trigger
...
       l1ttauRefs_.insert(l1ttauRefs_.end(),refs.begin(),refs.end());
       return l1ttauIds_.size();
     }
     size_type addObjects (const Vids& ids, const VRl1tetsum& refs) {
-      assert(ids.size()==refs.size());
```
Comment (Martin-Grunewald) [inline DataFormats/HLTReco/interface/TriggerRefsCollections.h]: Why do you remove the assert?

File: HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc [hunk 3]
```diff
@@ -121,6 +122,15 @@ TriggerSummaryAnalyzerRAW::analyze(const edm::Event& iEvent, const edm::EventSet
...
+       const unsigned int nL1TMuon(handle->l1tmuonSlice(iFO).second-
+				  handle->l1tmuonSlice(iFO).first);
+       if (nL1TMuon>0) LogVerbatim("TriggerSummaryAnalyzerRAW") << " L1TMuon: " << nL1TMuon;
+
+       const unsigned int nL1TEGamma(handle->l1tegammaSlice(iFO).second-
+				  handle->l1tegammaSlice(iFO).first);
+       if (nL1TEGamma>0) LogVerbatim("TriggerSummaryAnalyzerRAW") << " L1TEGamma: " << nL1TEGamma;
+
+
        LogVerbatim("TriggerSummaryAnalyzerRAW") << endl;
```
Comment (Martin-Grunewald) [inline HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc]: You added two new L1T types, but there should be 5.

File: HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc [hunk 4]
```diff
@@ -140,6 +150,8 @@ TriggerSummaryAnalyzerRAW::analyze(const edm::Event& iEvent, const edm::EventSet
      LogVerbatim("TriggerSummaryAnalyzerRAW") << "  PFJets:     " << handle->pfjetSize()     << endl;
      LogVerbatim("TriggerSummaryAnalyzerRAW") << "  PFTaus:     " << handle->pftauSize()     << endl;
      LogVerbatim("TriggerSummaryAnalyzerRAW") << "  PFMETs:     " << handle->pfmetSize()     << endl;
+     LogVerbatim("TriggerSummaryAnalyzerRAW") << "  L1TMuon:     " << handle->l1tmuonSize()    << endl;
+     LogVerbatim("TriggerSummaryAnalyzerRAW") << "  L1TEGamma:   " << handle->l1tegammaSize()    << endl;
```
Comment (Martin-Grunewald) [inline HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc]: You added two new L1T types, but there should be 5.

**Comment by Martin-Grunewald** [comment]
@rekovic 
Thanks for the update, and see comments above.

In addition, please see the template specialisation of methods
```TriggerSummaryProducerAOD::fillTriggerObject``` and
```TriggerSummaryProducerAOD::fillFilterObjectMember```.
Could you please see whether this is needed for the new L1T types?
I'd expect so at least for the new L1TEtSums, no?

File: DataFormats/HLTReco/interface/TriggerRefsCollections.h (hunk 1 above)
Comment (rekovic) [inline DataFormats/HLTReco/interface/TriggerRefsCollections.h]: Not intentional. Reverted. 2d0b01422d23781e09ad0ca74519b74bbd121548

File: DataFormats/HLTReco/interface/TriggerRefsCollections.h (hunk 2 above)
Comment (rekovic) [inline DataFormats/HLTReco/interface/TriggerRefsCollections.h]: Not intentional. Reverted. 2d0b01422d23781e09ad0ca74519b74bbd121548

File: HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc (hunk 3 above)
Comment (rekovic) [inline HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc]: Added missing 3. 2d0b01422d23781e09ad0ca74519b74bbd121548

File: HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc (hunk 4 above)
Comment (rekovic) [inline HLTrigger/HLTcore/plugins/TriggerSummaryAnalyzerRAW.cc]: Added missing 3.  2d0b01422d23781e09ad0ca74519b74bbd121548

**Comment by rekovic** [comment]
@Martin-Grunewald 
The above issues were addressed.  Thanks for spotting them.

As for the module TriggerSummaryProducerAOD, template specialisation is needed for L1EtSums, for sure, and is easily implemented.  I can work on this later, test it, and add to the PR when done.

**Comment by Martin-Grunewald** [comment]
@rekovic 
I assume this PR can be closed?!

**Pull request closed without merging**